conda_forge_tick/auto_tick.py

```diff
@@ -1303,26 +1302,32 @@ def _setup_limits():
 
 
 # @profiling
-def main(args: "CLIArgs") -> None:
+def main(
+    debug: bool = False,
+    dry_run: bool = False,
+    github_username: str = "",
+    github_password: str = "",
+    github_token: str = "",
+) -> None:
```

Comment:
Don't we need to change the old CLI functions to accommodate the change in function signature here?

Reply:
my intent with this PR is to have this replace the existing CLI with its integer-based invoking. It seems like an unusual pattern that isn't understandable without reading the code. This initial review has them separate for readability.

Reply:
if we want to keep the old CLI around then I can update the way they're invoked. Or I can make another layer of indirection and have `cli2` call into a corresponding wrapper and leave `main` alone. but the whole intent here is to update the CLI of this project with modern CLI infrastructure to simplify the code base not add another level of abstraction. Let me know your thoughts

Reply:
Right, my concern is only that if this PR breaks the old CLI, then we will break the bot since it runs live right off master. So we either need to do the work to start versioning things, or we need to make sure everything is backward compatible for at least a bit.

Reply:
ohhhh got it okay. so the new plan will be:

1. in this PR, add cli2.py and wrappers around each of the main() functions so that cli.py doesn't break
2. merge the bot PR that switches stuff to cli2.py
3. make another PR here that cleans up the old cli and gets rid of the shims. this pr should be transparent to the bot code

Reply:
That's perfect. We gotta do work on the truck while we're hauling ass at 70 MPH down the highway.